cloudferrylib/os/actions/vm_snapshot.py

```diff
@@ -91,16 +123,22 @@ class VmRestore(VmSnapshotBasic):
     def run(self, rollback_vars=None, *args, **kwargs):
         LOG.debug("restoring vms from snapshot")
         snapshot_from_namespace = kwargs.get(self.namespace_variable)
+        processing_vms = kwargs.get('processing_vms', [])
         compute = self.get_compute_resource()
         position = self.cloud.position
         vm_id_targets = ('src_id', 'dst_id') \
             if position == 'src' else ('dst_id', 'src_id')
+
         vms_succeeded = {}
         if rollback_vars:
             vms = rollback_vars.get('vms', [])
             vms_succeeded = {pair_vms[vm_id_targets[0]]: pair_vms[
                 vm_id_targets[1]] for pair_vms in vms}
-        for vm in self.get_list_of_vms():
+
+        search_opts_tenants = kwargs.get('search_opts_tenant',
+                                         {'all_tenant': True})
```

Comment:
Also you don't need to define all_tenants here, because `get_list_of_vms()` will do it anyway